euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Collector.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.core.client.io;
+
+/**
+ * Extends {@link Environment} with write capability. Used in
+ * user defined functors.
+ *
+ * @param <T> the type of elements collected through this context
+ */
+public interface Collector<T> extends Environment {
```

Comment:
I have a feeling this should have some other name, because besides _collecting_, it also provides access to window and possibly other stuff.